Added X property to author api endpoint

ronaldlangeveld opened the pull request:
ref https://linear.app/ghost/issue/PLG-415/support-using-x-in-place-of-twitter

- Using the serializer, I mapped an X property to have the same value as twitter in order to expose it so that it can be used as a condition (if statement) inside templates using handlebars, eg `{{#if x}}`.
- This change is reliant on https://github.com/TryGhost/SDK/pull/512

ronaldlangeveld committed:
Added x as authors endpoint

ref https://linear.app/ghost/issue/PLG-415/support-using-x-in-place-of-twitter

- Using the serliaizer, I mapped an X property to have the same value as
  twitter in order to expose it so that it can be used as a condition
(if statement) inside templates using handlebars, eg `{{#if x}}`.
- This change is reliant on https://github.com/TryGhost/SDK/pull/512

ErisDS commented on ghost/core/core/server/api/endpoints/utils/serializers/output/utils/clean.js:
```diff
@@ -68,6 +68,9 @@ const author = (attrs, frame) => {
         }
+
+        // if a twitter username is provided, use it as the x username to ensure backwards compatibility
+        attrs.x = attrs.twitter || null;
```
Although this pattern is not the greatest, the intention of the `clean.js` utility is for cleaning & removing attributes. There is an `extra-attrs.js` utility intended for adding attributes.

ErisDS commented on ghost/core/test/unit/frontend/helpers/social_url.test.js:
```diff
@@ -1,11 +1,11 @@
 const socialData = {
     facebook: 'testuser-fb',
     twitter: 'testuser-tw',
+    x: 'testuser-tw',
```
Whilst adding the new property here is correct with the new shape of the data, supporting x in social_urls will require additional changes to the helper (you'll need to map x to twitter) - I _think_ it makes sense to do those updates separately and focus in this PR on getting the API changes complete.

ronaldlangeveld replied:
I'll add these changes (tests) in a separate PR.

However, we can avoid making any changes to the `social_urls` template helper by creating an alias in the `@tryghost/social-urls` library in the SDK doc as I've done in this [PR](https://github.com/TryGhost/SDK/pull/512/files).

All that was needed for it work was to have `x` as a property in the user / author api and having `@tryghost/social-urls` export a function called `x`.

ronaldlangeveld committed:
Added x mapping to extra-attrs

ronaldlangeveld committed:
Fixed condition

ronaldlangeveld committed:
Update regressions snapshots

ronaldlangeveld committed:
Cleanup

ronaldlangeveld committed:
Added X mapping to `getSiteData()`

ErisDS commented on ghost/core/core/frontend/services/theme-engine/middleware/update-global-template-options.js:
```diff
@@ -16,6 +15,13 @@ function getSiteData() {
         siteData.signup_url = '#/portal';
     }
 
+    // theme only property added to @site
```
To me this is a discrete change again. I'd make it in a separate PR. Probably first as it's the simplest.

ronaldlangeveld committed:
Revert "Added X mapping to `getSiteData()`"

This reverts commit 1b2a3a7f0f729833596567b77f937c9b4494d3cb.

ronaldlangeveld committed:
Added X attribute to model instead of via extra-attrs

ref https://linear.app/ghost/issue/PROD-606/support-using-x-in-place-of-twitter

- Updated the X attribute to be in the Model instead as it's closer to
  the source of truth. Advised to do it like this during clean code
session.

ronaldlangeveld committed:
Merge branch 'main' into support-x-twitter

ronaldlangeveld committed:
Merge branch 'main' into support-x-twitter

ronaldlangeveld committed:
Merge branch 'main' into support-x-twitter
